### ISSUE TITLE
SessionAffinity in Service

### ISSUE THREAD

**Issue by costinm** [istio/ztunnel#450]
Another potential 'on by default' and 'do not break apps' issue: https://kubernetes.io/docs/reference/networking/virtual-ips/#session-affinity

In L4 mode - I think we may need some table to maintain the persistence. 
For Waypoints - this may need to use the header-based persistence filters in Envoy that we just added support for. 
Ztunnel would likely need to keep track of L7 Service with affinity and track and add the persistence header. 

Also we may consider using this affinity as another way to trigger header-based persistence with Sidecars and Gateways.

**Comment by costinm** [comment]
internalTrafficPolicy=Local is in the same category - may need a separate bug. Istiod knows the node - this one could be done in istiod-only ( not sure if we handle it for sidecars, we may want to fix it there as well but we run into the same 'what may break if w fix a bug' backward compatibiity issues ).

**Comment by howardjohn** [comment]
Agree, we also have https://github.com/istio/ztunnel/issues/204 tracking this more broadly. It will likely involve changing the XDS to have "Service" as a first class concept

**Comment by hzxuzhonghu** [comment]
This is also needed  for sidecar

**Comment by costinm** [comment]
I agree - anything that can be fixed or supported in sidecar as 'opt in'
and without backward compat is good to have.

But we can still do it first in ambient, and make sure the new model is
right - and slowly backport to sidecars too (
and first on gateways).
